vispy/visuals/image_complex.py

```diff
@@ -0,0 +1,96 @@
+from .image import ImageVisual, _apply_clim_float, _apply_gamma_float
+import numpy as np
+from .shaders import Function, FunctionChain
+
+# In a complex Image, the texture will be rg32f, where:
+# data.r contains the real component
+# data.g contains the imaginary component
+COMPLEX_TRANSFORMS = {
+    "real": "float cplx2float(vec4 data) { return data.r; }",
+    "imaginary": "float cplx2float(vec4 data) { return data.g; }",
+    "magnitude": "float cplx2float(vec4 data) { return length(data); }",
+    "phase": "float cplx2float(vec4 data) { return atan(data.g, data.r); }",
+}
+# aliases
+COMPLEX_TRANSFORMS["angle"] = COMPLEX_TRANSFORMS["phase"]
+COMPLEX_TRANSFORMS["abs"] = COMPLEX_TRANSFORMS["magnitude"]
+COMPLEX_TRANSFORMS["imag"] = COMPLEX_TRANSFORMS["imaginary"]
+
+
+class ComplexImageVisual(ImageVisual):
+    COMPLEX_MODES = set(COMPLEX_TRANSFORMS)
+
+    def __init__(self, data=None, complex_mode="magnitude", **kwargs):
+        if complex_mode not in self.COMPLEX_MODES:
+            raise ValueError(
+                "complex_mode must be one of %s" % ", ".join(self.COMPLEX_MODES)
+            )
+        self._data_is_complex = np.iscomplexobj(data)
+        self._complex_mode = complex_mode
+
+        if kwargs.get("clim", "auto") == "auto" and self._data_is_complex:
+            kwargs["clim"] = self._calc_complex_clim(data)
+
+        kwargs["texture_format"] = "auto"
+        super().__init__(data=data, **kwargs)
+
+    def set_data(self, image):
+        data = np.asarray(image)
+        self._data_is_complex = np.iscomplexobj(data)
+        if self._data_is_complex:
+            #  Turn the texture into an rg32f texture
+            # where r = 'real' and g = 'imag'
+            self._texture._format = "rg"
+            data = np.stack([data.real, data.imag], axis=-1)
```

Comment:
I think we can do this without data-copies, because the complex and imaginary part are aligned similar to the color planes in an RGB image. I think this should work: 
```py
data = data.astype(np.complex64, copy=False)  # turn complex128 into complex64 if needed
data = data.view(dtype=np.float32).reshape((data.shape + (2, ))) 
```